other_data_collections/2015ApJ...812...60B/biteau.py

```diff
@@ -1,46 +1,177 @@
+    else:
+        filename='tev-' + str(source_id) + '-sed.ecsv'
+    if(experiment == 'MAGIC'):
+        new_table.meta['telescope'] = 'magic'
+    elif(experiment == 'CAT'):
+        new_table.meta['telescope'] = 'cat'
+    elif(experiment == 'HEGRA'):
+        new_table.meta['telescope'] = 'hegra'
+    elif(experiment == 'TIBET'):
+        new_table.meta['telescope'] = 'tibet'
+    elif(experiment == 'HESS'):
+        new_table.meta['telescope'] = 'hess'
+    elif(experiment == 'TACTIC'):
+        new_table.meta['telescope'] = 'tactic'
+    elif(experiment == 'ARGO-YBJ'):
+        new_table.meta['telescope'] = 'argo'
+    elif(experiment == 'VERITAS'):
+        new_table.meta['telescope'] = 'veritas'
+    elif(experiment == 'Whipple'):
+        new_table.meta['telescope'] = 'whipple'
+    new_table.meta.pop('author')
+    new_table.meta.pop('dataset_reference')
+    new_table.meta.pop('creation_date')
+    new_table.meta['filecounter'] = filecounter
+    new_table.meta['source_id'] = int(source_id)
+    new_table.meta['reference_id'] = reference_id
+    new_table.meta['note'] = note
+    new_table.meta['data_type'] = 'sed'
+    print('Length of table: {}'.format(len(new_table)))
+    new_table.write(filename, format='ascii.ecsv', delimiter=' ')
 
-        self.table.show_in_browser(jsviewer=True)
-        self.table2.show_in_browser(jsviewer=True)
+def create_escv2(table, experiment, reference_id, source, source_id):
+    new_table = Table(names=('e_ref', 'dnde', 'dnde_errn', 'dnde_errp'), dtype=('float32', 'float32', 'float32', 'float32'))
 
-    def run_checks(self):
-        # self.table.info('stats')
-        # self.table.pprint()
-        # ref_id = self.table['reference_id']
-        table = self.table.group_by('reference_id')
-        table.groups.keys.pprint(max_lines=-1)
+    new_table.meta = table.meta
 
-        # Note: all references are valid ADS bibcodes now!
-        # reference_ids = list(table.groups.keys['reference_id'])
-        # for reference_id in reference_ids:
-        #     article = list(ads.SearchQuery(q='bibcode:{}'.format(reference_id)))[0]
-        #     print(reference_id, article)
+    print('arguments of function:')
+    print(source)
+    print(reference_id)
+    print(experiment)
+    print('\n')
+    print('Biteau Catalog (For debug):')
+    print(table['source'][550])
+    print(table['reference_id'][550])
+    print('{} \n'.format(table['experiment'][550]))
 
-        # import IPython; IPython.embed()
-
-
-if __name__ == '__main__':
-    maker = BiteauMaker()
-    # biteau.run_basic_checks()
-    maker.run_checks()
+    for i in range(0, len(table)):
+        if((table['source'][i]==source) and (table['reference_id'][i] == reference_id) and (table['experiment'][i] == experiment)):
+            new_table.add_row((4.135667662E-27*table[i]['freq'], table[i]['e2dnde']/1.6022, table[i]['e2dnde_errn']/1.6022, table[i]['e2dnde_errp']/1.6022))
```

Comment:
This is not well readable.
Please use `astropy.units` or at least given your magic constants like 4.1e-27 and 1.6 names.
And wrap the line so that it's ~ 80 char max width, so that the code becomes readable without having to scroll back and forth.
Maybe a `row = table[i]` would also help (instead of using `[i]` everywhere in the formula that does the flux computation.